ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -442,18 +425,44 @@ private void validateQueries(final Map<PreparedStatement<?>, QueryMetadata> quer
   private void registerQueries(final List<QueryMetadata> queries) {
     for (final QueryMetadata queryMetadata : queries) {
       if (queryMetadata instanceof PersistentQueryMetadata) {
-        livePersistentQueries.add(queryMetadata);
-        final PersistentQueryMetadata persistentQueryMd = (PersistentQueryMetadata) queryMetadata;
-        persistentQueries.put(persistentQueryMd.getQueryId(), persistentQueryMd);
-        metaStore.updateForPersistentQuery(persistentQueryMd.getQueryId().getId(),
-            persistentQueryMd.getSourceNames(),
-            persistentQueryMd.getSinkNames());
+        final PersistentQueryMetadata persistentQuery = (PersistentQueryMetadata) queryMetadata;
+        persistentQueries.put(persistentQuery.getQueryId(), persistentQuery);
+        metaStore.updateForPersistentQuery(persistentQuery.getQueryId().getId(),
+            persistentQuery.getSourceNames(),
+            persistentQuery.getSinkNames());
       }
       allLiveQueries.add(queryMetadata);
     }
     engineMetrics.registerQueries(queries);
   }
 
+  private void unregisterQuery(final QueryMetadata query) {
+    final String applicationId = query.getQueryApplicationId();
+
+    if (!query.getState().equalsIgnoreCase("NOT_RUNNING")) {
```

Comment:
not really for this but wondering why we return a `String` from `QueryMetadata::getState` when it is getting an Enum from streams?

Reply:
Just so the use of streams is encapsulated within the class, i.e. no exposed through the API.  I could of converted to a non-streams enum, but didn't see the point.